tests/test_linux/fixtures/dhcp_servers/mock.py

```diff
@@ -0,0 +1,58 @@
+import asyncio
+
+import pytest
+from fixtures.pcap_files import PcapFile
+
+from pyroute2.dhcp.dhcp4socket import AsyncDHCP4Socket
+from pyroute2.dhcp.messages import SentDHCPMessage
+
+
+class MockDHCPServerFixture:
+    '''A fixture that's used to avoid a real dhcp server in unit tests.
+
+    By replacing `AsyncDHCP4Socket.loop` with an instance of this class,
+    it will wait for messages sent by `AsyncDHCPClient` and answer
+    with data from `responses`, in order.
+
+    The requests made by the client will be stored in `decoded_requests`.
+    '''
+
+    def __init__(
+        self, event_loop: asyncio.AbstractEventLoop, responses: list[bytes]
+    ):
+        self.responses: list[bytes] = responses
+        self.requests: list[bytes] = []
+        self.decoded_requests: list[SentDHCPMessage] = []
+        # save the event_loop fixture, 'cause pytest-asyncio on Python < 3.10
+        # drops the loop in the middle of the test
+        self.loop = event_loop
```

Comment:
@etene that did the trick

I've learned something new during this brainstorm :)